pkg/cloudevents/transport/http/options_test.go

```diff
@@ -351,6 +354,55 @@ func TestWithPort(t *testing.T) {
 	}
 }
 
+// Force a transport to close its server/listener by cancelling StartReceiver
+func forceClose(tr *Transport) {
+	ctx, cancel := context.WithCancel(context.Background())
+	go func() { tr.StartReceiver(ctx) }()
+	cancel()
+}
+
+func TestWithPort0(t *testing.T) {
+	testCases := map[string]func() (*Transport, error){
+		"WithPort0": func() (*Transport, error) { return New(WithPort(0)) },
+		"SetPort0":  func() (*Transport, error) { return &Transport{Port: new(int)}, nil },
+	}
+	for name, f := range testCases {
+		t.Run(name, func(t *testing.T) {
+			tr, err := f()
+			if err != nil {
+				t.Fatal(err)
+			}
+			defer func() { forceClose(tr) }()
+			port := tr.GetPort()
+			if port <= 0 {
+				t.Error("no dynamic port")
+			}
+			if d := cmp.Diff(port, *tr.Port); d != "" {
+				t.Error(d)
+			}
+		})
+	}
+}
+
+func TestWithListener(t *testing.T) {
+	l, err := net.Listen("tcp", ":0")
```

Comment:
how is this different than `SetPort(0)` ?

Reply:
I can say net.Listen("tcp6", "myhost:0") to listen for IPv6 connections only, and only on the interface identfied by myhost. This is a typical pattern for packages that listen: there may be a short cut  (e.g. net.http.ListenAndServe() but it's usually possible for the caller to provide the Listener for flexibility  (e.g. net.http.Serve(Listener))